In [2]:
import torch
import torch.nn as nn
class MyModel(nn.Module):
    def __init__(self, num_input_features, num_hidden_neuron, num_output_neuron):
        super(MyModel, self).__init__()
        self.hidden_layer = nn.Linear(num_input_features, num_hidden_neuron)
        self.output_layer = nn.Linear(num_hidden_neuron, num_output_neuron)
        self.tanh = nn.Tanh()
        self.sigmoid = nn.Sigmoid()
        
    def my_tanh(self,value):
        return (torch.exp(value) - torch.exp(-value))/(torch.exp(value) + torch.exp(-value))
    
    def my_sigmoid(self,value):
        return 1/(1 + torch.exp(-value))

    def forward(self, X):
        X = torch.tensor(X, dtype=torch.float)
        hidden_res = self.my_tanh(self.hidden_layer(X))
        output = self.my_sigmoid(self.output_layer(hidden_res))
        return output

In [3]:
#4.1
torch.manual_seed(1)
x = [[1,2,3],[4,5,6]]
neuralNetwork = MyModel(len(x[0]), 50, 1)
result = neuralNetwork.forward(x)
print("result : ",result)

result :  tensor([[0.4892],
        [0.5566]], grad_fn=<MulBackward0>)


when i used torch tanh and sigmoid func, result = tensor([[0.4892],[0.5566]], grad_fn=<SigmoidBackward0>)
when i used my own tanh and sigmoid func, result = tensor([[0.4892],[0.5566]], grad_fn=<MulBackward0>)
then my own functions is working correctly

In [4]:
#4.2
torch.manual_seed(200401093)
x = [[1,2,3],[4,5,6]]
neuralNetwork = MyModel(len(x[0]), 50, 1)
result = neuralNetwork.forward(x)
print("result : ",result)

result :  tensor([[0.6684],
        [0.7161]], grad_fn=<MulBackward0>)
